src/core/main.c

```diff
@@ -2449,6 +2457,14 @@ static void reset_arguments(void) {
         arg_random_seed_size = 0;
         arg_clock_usec = 0;
 
+        arg_mount_rate_limit_burst = DEFAULT_MOUNT_RATE_LIMIT_BURST;
+        const char *e = secure_getenv("SYSTEMD_DEFAULT_MOUNT_RATE_LIMIT_BURST");
```

Comment:
reset_arguments() is supposed to reset everything to defaults, i.e. the  secure_getenv() check and parsing should be moved out of here and instead into parse_config_file()